Zend/zend_execute_API.c

```diff
@@ -396,6 +423,17 @@ void shutdown_executor(void) /* {{{ */
 	bool fast_shutdown = is_zend_mm() && !EG(full_tables_cleanup);
 #endif
 
+#if defined(ZTS) && defined(HAVE_TIMER_CREATE)
+	if (timer_delete(EG(timer)) != 0) {
```

Comment:
This should be in zend_unset_timeout()

Reply:
The timer_delete() can be called in the function passed to tsrm_set_shutdown_handler(), although this will only be called once for all threads.

Reply:
I'll work on this tomorrow.